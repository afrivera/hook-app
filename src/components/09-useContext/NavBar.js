import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
                <Link to="/" className="navbar-brand" >UseContext</Link>           
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink exact="true" activeClassName="active" to="/" className="nav-link" aria-current="page" >Home</NavLink>
                        <NavLink exact="true" activeClassName="active" to="/About" className="nav-link" >About </NavLink>
                        <NavLink exact="true" activeClassName="active" to="/Login" className="nav-link" >Login</NavLink>
                    </div>
                </div>
        </nav>
    )
}
