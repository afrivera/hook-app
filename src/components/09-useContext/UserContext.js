import {createContext} from 'react';

export const UsertContext = createContext(null)
