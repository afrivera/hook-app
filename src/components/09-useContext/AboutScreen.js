import React, { useContext } from 'react';
import { UsertContext } from './UserContext';

export const AboutScreen = () => {

    const {user,setUser} = useContext(UsertContext);

    const handleClick = () =>{
        setUser({});
    }

    return (
        <div>
            <h1>About Screen</h1>
            <hr/> 

            <pre>
                {JSON.stringify(user,null,3)}
            </pre>
            
            <button
                className="btn btn-warning"
                onClick={handleClick}
            >
                Logout
            </button>
        </div>
    )
}
