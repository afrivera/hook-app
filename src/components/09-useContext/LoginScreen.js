import React, { useContext } from 'react'
import { UsertContext } from './UserContext'

export const LoginScreen = () => {

    //1. obtener la referencia al usercontext
    const {setUser} = useContext(UsertContext);

    //2. setUser
    /* 
        {
            id:123,
            name: 'Andres'
        }
    */

    return (
        <div>
            <h1>Login Screen</h1>
            <hr/>     
            <button
                className="btn btn-primary"
                onClick={ () =>setUser( {
                    id:123,
                    name: 'Andres'
                })}
            >
                Login    
            </button>        
        </div>
    )
}
