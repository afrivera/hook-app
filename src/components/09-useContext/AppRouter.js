import React from 'react';
import {
    BrowserRouter as Router,
    Route,
    Routes,
} from "react-router-dom";
import { NavBar } from './NavBar';

import {AboutScreen} from './AboutScreen';
import {LoginScreen} from './LoginScreen';
import {HomeScreen} from './HomeScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <div className="container">

                <Routes>
                    <Route exact path="/about" element={<AboutScreen />} />
                    <Route exact path="/login" element={<LoginScreen />} />
                    <Route exact path="/" element={<HomeScreen />} />
                    <Route path="*" element={<HomeScreen/>} />
                </Routes>
                </div>
            </div>
        </Router>
    )
}
