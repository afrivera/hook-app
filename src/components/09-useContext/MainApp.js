import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UsertContext } from './UserContext'



export const MainApp = () => {

    const [user, setUser] = useState({});
    return (
        <UsertContext.Provider value={{
            user,
            setUser
        }}>
            <AppRouter />

        </UsertContext.Provider>            
        
    )
}
