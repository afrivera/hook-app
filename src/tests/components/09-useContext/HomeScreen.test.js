import { mount } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UsertContext } from '../../../components/09-useContext/UserContext';


describe('pruebas en <HomeScreen />', () => {

    const user = {
        name: 'Andres',
        email: 'andres@gmail.com'
    }

    //el shallow solo evalua el componente principal o el primero
    const wrapper = mount(
        <UsertContext.Provider value={{
            user,

        }}>
            <HomeScreen />

        </UsertContext.Provider>
    );
    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    
})
