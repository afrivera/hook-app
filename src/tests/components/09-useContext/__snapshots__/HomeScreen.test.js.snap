// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <HomeScreen /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Andres",
   "email": "andres@gmail.com"
}
  </pre>
</div>
`;
